### PR TITLE
Enable copy & paste

### PR DESCRIPTION
**Pull request by wehrlean**
[Trello](https://trello.com/c/zmtSBCrI/482-enable-copypaste)

### CODE REVIEW

File: web/src/app/modules/views/main/editors/modules/graphical-editor/components/graphical-editor.component.ts [hunk 1]
```diff
@@ -187,6 +187,32 @@ export class GraphicalEditor {
...
+
+        // Ctrl+X
+        this.keyHandler.bindControlKey(88, (evt: KeyboardEvent) => {
+          this.stopEvent(evt);
+          if (this.graph.isEnabled()) {
+            mx.mxClipboard.copy(this.graph, this.graph.getSelectionCells());
+            this.deleteSelectedCells();
+
+          }
+        });
```
Comment (tobi321) [inline web/src/app/modules/views/main/editors/modules/graphical-editor/components/graphical-editor.component.ts]: Add this to support command button (equivalent to STRG button on Windows keyboards) when using a Mac Keyboard

`        this.keyHandler.getFunction = function(evt){
            if (evt != null) {
                return (mx.mxEvent.isControlDown(evt) || (mx.mxClient.IS_MAC && evt.metaKey)) ? this.controlKeys[evt.keyCode] : this.normalKeys[evt.keyCode];
            }
        return null;
        };`

**Review by tobi321 — CHANGES_REQUESTED** [review]
Nice! After including my code from the comment, maybe you can check if it still works on Windows. 

If I select a single node and try to copy it, it works. 
However, if I select multiple nodes with STRG + Click and then try to copy them it does not work. I can only copy multiple parts of the CEG when I either span a selection window with the mouse over the nodes or if I press STRG + A.

File: web/src/app/modules/views/main/editors/modules/graphical-editor/components/graphical-editor.component.ts (hunk 1 above)
Comment (wehrlean) [reply web/src/app/modules/views/main/editors/modules/graphical-editor/components/graphical-editor.component.ts]: I'm not sure if I added it in the right way. Can't test it on Windows

**Comment by wehrlean** [comment]
> Nice! After including my code from the comment, maybe you can check if it still works on Windows.
> 
> If I select a single node and try to copy it, it works.
> However, if I select multiple nodes with STRG + Click and then try to copy them it does not work. I can only copy multiple parts of the CEG when I either span a selection window with the mouse over the nodes or if I press STRG + A.


Selecting multiples nodes with STRG + Click work in my case. Please test it a second time

File: web/src/app/modules/views/main/editors/modules/graphical-editor/components/graphical-editor.component.ts [hunk 2]
```diff
@@ -22,344 +22,367 @@ import { StyleChanger } from './util/style-changer';
...
+    this.keyHandler.getFunction = function (evt) {
+      if (evt != null) {
+        return (mx.mxEvent.isControlDown(evt) || (mx.mxClient.IS_MAC && evt.metaKey))
+          ? this.controlKeys[evt.keyCode]
+          : this.normalKeys[evt.keyCode];
+      }
+      return null;
+    };
+
+    /* mx.mxEvent.addListener(this.graph.container, mx.mxEvent.CLICK, function () {
```
Comment (tobi321) [inline web/src/app/modules/views/main/editors/modules/graphical-editor/components/graphical-editor.component.ts]: Remove commented-out code

**Review by tobi321 — CHANGES_REQUESTED** [review]
Nice, now the selection of multiple nodes and copy&paste actions are bind to the correct keys on a mac keyboard!

**Review by tobi321 — APPROVED** [review]

**Pull request merged**